src/V3Randomize.cpp

```diff
@@ -141,14 +150,25 @@ class RandomizeMarkVisitor final : public VNVisitorConst {
 //######################################################################
 // Visitor that turns constraints into template strings for solvers
 
+static AstScope* findClassScope(AstClass* classp) {
+    AstScope* scopep = nullptr;
+    for (AstNode* childp = classp->stmtsp(); childp != nullptr; childp = childp->nextp()) {
+        scopep = VN_CAST(childp, Scope);
+        if (scopep) break;
+    }
```

Comment:
This can be slow and I see it called many places - can you memoize the scope for a given class in a user() or use a std::map?